test/unit/crout/crout.cpp

```diff
@@ -0,0 +1,114 @@
+/*************************************************************************
+ * Copyright (C) 2018-2019 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#define CATCH_CONFIG_MAIN
+
+#include <chrono>
+#include <random>
+
+#include <catch/catch.hpp>
+
+#include "Eigen/Dense"
+#include "Eigen/LU"
+
+#include "nmodl.hpp"
+
+using namespace nmodl;
+using namespace Eigen;
+using namespace std;
+
+
+/// https://stackoverflow.com/questions/15051367/how-to-compare-vectors-approximately-in-eigen
+template <typename DerivedA, typename DerivedB>
+bool allclose(const Eigen::DenseBase<DerivedA>& a,
+              const Eigen::DenseBase<DerivedB>& b,
+              const typename DerivedA::RealScalar& rtol =
+                  Eigen::NumTraits<typename DerivedA::RealScalar>::dummy_precision(),
+              const typename DerivedA::RealScalar& atol =
+                  Eigen::NumTraits<typename DerivedA::RealScalar>::epsilon()) {
+    return ((a.derived() - b.derived()).array().abs() <= (atol + rtol * b.derived().array().abs()))
+        .all();
+}
+
+
+template <typename T>
+bool test_Crout_correctness(T rtol = 1e-6, T atol = 1e-6) {
```

Comment:
these tolerances are quite big. why?

Reply:
@cattabiani you mean that we can relax them even more (like 1e-3)?

Reply:
Yes, I meant what Ohm said

Reply:
@ohm314 & @cattabiani : I have modified Crout solver and I have resolved an accuracy issue that I had. I computed the relative error (A*x-b)/b for both eigen and crout, and crout's relative error was consistently two orders of magnitude higher than eigen one (before the modification). Therefore, for small tolerances my tests where failing. Now (after the modification), the relative error for both solvers has the same order of magnitude.